operator/v1/types_console.go

```diff
@@ -55,15 +55,22 @@ type ConsoleSpec struct {
 	// +optional
 	Route ConsoleConfigRoute `json:"route"`
 	// plugins defines a list of enabled console plugin names.
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=64
+	// +kubebuilder:validation:XValidation:rule="self.all(x, self.exists_one(y, x == y))",message="each plugin name must be unique"
 	// +optional
-	Plugins []string `json:"plugins,omitempty"`
+	Plugins []PluginName `json:"plugins,omitempty"`
 	// ingress allows to configure the alternative ingress for the console.
 	// This field is intended for clusters without ingress capability,
 	// where access to routes is not possible.
 	// +optional
 	Ingress Ingress `json:"ingress"`
 }
 
+// +kubebuilder:validation:MaxLength=128
+// +kubebuilder:validation:Pattern=^[a-zA-Z0-9-]+$
```

Comment:
A couple things:
- There didn't appear to be any pattern validation on the value previously. Why was this pattern chosen? Would introducing this pattern result in plugins that are no longer able to be used?
- Conventions are shifting away from using the `kubebuilder:validation:Pattern` marker in favor of writing a CEL validation, using the `kubebuilder:validation:XValidation` marker, to provide users a more clear message when their input is invalid. If this pattern stays, update this to use a CEL validation instead?